### DISCUSSION TITLE
WASM for apps and watch faces?

### DISCUSSION THREAD

**Discussion by spaghetus** [InfiniTimeOrg/InfiniTime#1823] [Ideas]
I read in #1738 that installing apps at runtime is on the roadmap, and I think using WASM would be a nice way to do it. It would allow app developers to use many different languages, and I was able to find a [relatively fast interpreter](https://github.com/wasm3/wasm3) written in C++ that InfiniTime could depend on.

Performance probably wouldn't be ideal if someone tries to make an app with Python or something, but for native languages like C++ or Rust I would expect it to be good enough.

**Comment by minacode** [comment]
> Minimum useful system requirements: ~64Kb for code and ~10Kb RAM

We don't have this, sorry 😕

**Comment by pipe01** [comment]
I looked into this a while ago but my conclusion was that the WASM runtime and its overhead was too big and it would be very difficult to fit on such small constraints as InfiniTime has. I've instead been working on a small bytecode language [here](https://github.com/InfiniTimeOrg/InfiniTime/pull/2092) that can be used for making watchfaces and requires little resources.
